### PR TITLE
Doxygen fixes2

### PR DESCRIPTION
**Pull request by kugel-**
Separated fixes from https://github.com/geany/geany/pull/744

I made more changes this time around as a result of the discussion there (and I found some more minor issues).

### CODE REVIEW

File: src/filetypes.h [hunk 1]
```diff
@@ -106,7 +109,9 @@ typedef enum
 	/* ^ append items here */
 	GEANY_MAX_BUILT_IN_FILETYPES	/* Don't use this, use filetypes_array->len instead */
 }
-filetype_id;
+GeanyFiletypeID;
+
+#define filetype_id GeanyFiletypeID /* compat define - should be removed in the future */
```
Comment (b4n) [inline src/filetypes.h]: this is not really related, but I guess it's good

**Comment by b4n** [comment]
I guess it looks mostly fine now

**Comment by kugel-** [comment]
Cool, please merge :-) Feel free to squash before merging

**Comment by kugel-** [comment]
What's holding this back?

File: src/encodings.h [hunk 2]
```diff
@@ -40,29 +40,13 @@
...
-	NONE = 0,
-	WESTEUROPEAN,
-	EASTEUROPEAN,
-	EASTASIAN,
-	ASIAN,
-	MIDDLEEASTERN,
-	UNICODE,
-
-	GEANY_ENCODING_GROUPS_MAX
-} GeanyEncodingGroup;
```
Comment (b4n) [inline src/encodings.h]: @codebrainz this breaks GeanyPy

File: src/encodings.h [hunk 3]
```diff
@@ -139,23 +123,8 @@ typedef enum
...
-	gint   				 idx;
-	/** Internally used member for grouping */
-	gint   				 order;
-	/** Internally used member for grouping */
-	GeanyEncodingGroup   group;
-	/** String representation of the encoding, e.g. "ISO-8859-3" */
-	const gchar			*charset;
-	/** Translatable and descriptive name of the encoding, e.g. "South European" */
-	const gchar			*name;
-} GeanyEncoding;
```
Comment (b4n) [inline src/encodings.h]: @codebrainz this breaks GeanyPy too.

Both are trivial to fix tho (just remove the wrappers).  Technically this struct *was* part of the API, but as there was nothing to get any instance of it removing the struct itself makes sense.

Comment (kugel-) [inline src/encodings.h]: talked to @codebrainz on IRC about both. We agreed on fixing geanypy.
